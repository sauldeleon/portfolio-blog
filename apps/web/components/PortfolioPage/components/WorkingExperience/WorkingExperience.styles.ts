import { styled } from 'styled-components'

import { Heading, Label } from '@sdlgr/typography'

export const StyledCompanyName = styled(Heading)`
  ${({ theme }) => theme.fontStyles.robotoMono.medium};
  margin-bottom: 30px;
`

export const StyledCompanyPeriod = styled(Label)`
  font-size: 20px;
  margin-bottom: 20px;
  ${({ theme }) => theme.fontStyles.robotoMono.regular};
`

export const StyledList = styled.ul`
  margin-bottom: 50px;
`

export const StyledListItem = styled.li`
  font-size: 16px;
  text-indent: -20px;
  padding-left: 20px;

  &:before {
    content: 'Â·';
    font-size: 30px;
    vertical-align: middle;
    line-height: 20px;
  }
`

export const StyledTechnology = styled.b`
  ${({ theme }) => theme.typography.body.M}
  font-size: 16px;
  font-weight: bold;
`
