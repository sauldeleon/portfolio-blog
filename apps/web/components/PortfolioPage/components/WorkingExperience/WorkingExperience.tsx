import { format } from 'date-fns/format'
import { useId } from 'react'
import { Trans } from 'react-i18next'

import { useExperienceItems } from '@web/components/ExperiencePage/useExperienceItems'
import { useClientTranslation } from '@web/i18n/client'

import {
  StyledCompanyName,
  StyledCompanyPeriod,
  StyledList,
  StyledListItem,
  StyledTechnology,
} from './WorkingExperience.styles'

export function WorkingExperience() {
  const experienceItems = useExperienceItems()
  const id = useId()
  const { t } = useClientTranslation('experiencePage')

  return (
    <>
      {experienceItems.map(
        ({ order, company, beginDate, endDate, descriptionParagraphKeys }) => (
          <div key={order}>
            <StyledCompanyName $level={2}>
              {company} - {t(descriptionParagraphKeys[0])}
            </StyledCompanyName>
            <StyledCompanyPeriod>
              {`${format(beginDate, 'MMMM yyyy')}${
                endDate ? format(endDate, ' - MMMM yyyy') : ''
              }`}
            </StyledCompanyPeriod>
            <StyledList>
              {descriptionParagraphKeys.slice(1).map((paragraphKey, index) => (
                <StyledListItem key={`${id}-${index}`}>
                  <Trans
                    t={t}
                    i18nKey={paragraphKey}
                    components={{
                      bold: <StyledTechnology />,
                    }}
                  />
                </StyledListItem>
              ))}
            </StyledList>
          </div>
        ),
      )}
    </>
  )
}
