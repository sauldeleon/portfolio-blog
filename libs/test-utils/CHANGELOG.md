# Changelog Test utils

## [0.2.3](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.2...test-utils-0.2.3) (2024-01-08)

## [0.2.2](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.1...test-utils-0.2.2) (2023-09-26)

## [0.2.1](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.0...test-utils-0.2.1) (2023-09-14)

# [0.2.0](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.1.0...test-utils-0.2.0) (2023-09-05)

### Bug Fixes

- reduced duplicated lines ([f51168a](https://github.com/sauldeleon/portfolio-blog/commit/f51168aefe025a5a65ffdb2dc22f7a16b863406e))
- rename commit message format ([e1e2ab4](https://github.com/sauldeleon/portfolio-blog/commit/e1e2ab404bbd2c32f3508d1ed8197b3fbff93cb9))

### Features

- deleted changelogs ([e9be338](https://github.com/sauldeleon/portfolio-blog/commit/e9be33836ee47b6505ad94d21f4be21855a7fa0d))

# 0.1.0 (2023-09-02)

### Bug Fixes

- rename commit message format ([e1e2ab4](https://github.com/sauldeleon/portfolio-blog/commit/e1e2ab404bbd2c32f3508d1ed8197b3fbff93cb9))

### Features

- add contact initial page ([fea862b](https://github.com/sauldeleon/portfolio-blog/commit/fea862b3507268013149050ae3f05fa517e7b3d9))
- add homepage ([60523aa](https://github.com/sauldeleon/portfolio-blog/commit/60523aa31ec088d318745ef0c757b4cbe656f0d4))
- deleted changelogs ([e9be338](https://github.com/sauldeleon/portfolio-blog/commit/e9be33836ee47b6505ad94d21f4be21855a7fa0d))
- improved a11y and styles ([873f656](https://github.com/sauldeleon/portfolio-blog/commit/873f656587f50df8e29ad526fdc68fea0b2c9784))
- toggle language button ([afd97e9](https://github.com/sauldeleon/portfolio-blog/commit/afd97e95beac0f70ed643c753e71db2ca014054b))
