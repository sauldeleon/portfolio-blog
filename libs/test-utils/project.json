{
  "name": "test-utils",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/test-utils/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "version": {
      "executor": "@jscutlery/semver:version",
      "options": {
        "changelogHeader": "# Changelog Test utils \n",
        "baseBranch": "main",
        "commitMessageFormat": "chore(${projectName}): release: v${version}",
        "push": false
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/test-utils/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/test-utils/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    }
  }
}
