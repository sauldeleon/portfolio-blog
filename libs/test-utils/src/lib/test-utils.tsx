import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { Matcher } from '@testing-library/dom/types/matches'
import {
  RenderOptions,
  RenderResult,
  queryAllByAttribute as _queryAllByAttribute,
  queryByAttribute as _queryByAttribute,
  act,
  render,
  screen,
} from '@testing-library/react'
import { RouterContext } from 'next/dist/shared/lib/router-context'
import Router, { NextRouter } from 'next/router'
import { ReactNode, useEffect, useState } from 'react'
import { ThemeProvider } from 'styled-components'

import { LanguageContextProvider } from '@sdlgr/i18n-config'
import { mainTheme } from '@sdlgr/main-theme'

export type MatcherCustomFnParams = {
  content: string
  element: Element | null
}

/* istanbul ignore next */
export const renderWithTheme = (children: ReactNode): RenderResult => {
  return render(<ThemeProvider theme={mainTheme}>{children}</ThemeProvider>)
}

/* istanbul ignore next */
export function RenderProviders({ children }: { children: ReactNode }) {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  })

  return (
    <QueryClientProvider client={queryClient}>
      <LanguageContextProvider value={{ language: 'en' }}>
        <ThemeProvider theme={mainTheme}>{children}</ThemeProvider>
      </LanguageContextProvider>
    </QueryClientProvider>
  )
}

/* istanbul ignore next */
export const renderApp = (
  children: ReactNode,
  options?: Omit<RenderOptions, 'queries'>
) => {
  return render(<RenderProviders>{children}</RenderProviders>, options)
}

/**
 * Generic function to get element by attribute and value
 * @example queryByAttribute('id')(baseElement, 'element-id')
 * @param {string} attr
 * @return {any}
 */
export function queryByAttribute(attr: string) {
  return _queryByAttribute.bind(null, attr)
}

/**
 * Generic function to get all elements by attribute and value
 * @example queryByAttribute('class')(baseElement, 'element-class')
 * @param {string} attr
 * @returns {any}
 */
export function queryAllByAttribute(attr: string) {
  return _queryAllByAttribute.bind(null, attr)
}

/**
 * Generic function to get element by attribute and value
 * @example getByAttribute('id')(baseElement, 'element-id')
 * @param {string} attr
 * @return {any}
 */
export function getByAttribute(attr: string) {
  return _queryByAttribute.bind(null, attr)
}

/**
 * Generic function to get all elements by attribute and value
 * @example getByAttribute('class')(baseElement, 'element-class')
 * @param {string} attr
 * @returns {any}
 */
export function getAllByAttribute(attr: string) {
  return _queryAllByAttribute.bind(null, attr)
}

/**
 * Shortcut version for queryByAttribute, but specific for id
 * @param {HTMLElement} container
 * @param {string} id
 * @return {any}
 */
export function getById(container: HTMLElement, id: string) {
  return queryByAttribute('id')(container, id) as Element
}

/**
 * Shortcut version for queryByAttribute, but specific for class name
 * @param {HTMLElement} container
 * @param {string} className
 * @return {any}
 */
export function getByClassName(container: HTMLElement, className: string) {
  return queryByAttribute('class')(container, className) as Element
}

function matcherCustomFn(
  tag: string,
  matcher: Matcher,
  { element, content }: MatcherCustomFnParams
) {
  const isTag = element?.tagName.toUpperCase() === tag.toUpperCase()
  let matching = false
  if (isTag) {
    matching = content === matcher
    /* istanbul ignore else*/
    if (typeof matcher === 'function') {
      matching = matcher(content, element)
    } else if (matcher instanceof RegExp) {
      matching = matcher.test(content)
    }
  }
  return matching && isTag
}

/**
 * Get 1 element by tag and text
 * @param {string} tag
 * @param {Matcher} matcher
 * @return {any}
 */
export function getByTagAndText(tag: string, matcher: Matcher) {
  return screen.getByText((content, element) =>
    matcherCustomFn(tag, matcher, { content, element })
  )
}

/**
 * Get multiple elements by tag and text
 * @param {string} tag
 * @param {Matcher} matcher
 * @return {any}
 */
export function getAllByTagAndText(tag: string, matcher: Matcher) {
  return screen.getAllByText((content, element) =>
    matcherCustomFn(tag, matcher, { content, element })
  )
}

/**
 * Mock Math.random() to return a specific value
 * @param {number} value
 */
export function mockRandom(value = 0.1234) {
  const mockMath = Object.create(global.Math)
  mockMath.random = () => value
  global.Math = mockMath
}

/**
 * Hook to use in test to ensure "last" render is called
 */
export function useWithRefreshInTest() {
  const [withRefreshInTest, setWithRefreshInTest] = useState(false)

  useEffect(() => {
    // Change state to "rerender" the component
    setWithRefreshInTest(true)
  }, [])

  return withRefreshInTest
}

export { render } from '@testing-library/react'

/* istanbul ignore next */
export const awaitAllAsync = async () =>
  act(() => new Promise((resolve) => setTimeout(resolve, 0)))

/* istanbul ignore next */
export function renderWithNextRouter(
  children: ReactNode,
  options?: Partial<NextRouter>
) {
  return renderWithTheme(
    <RouterContext.Provider value={{ ...Router, ...options }}>
      {children}
    </RouterContext.Provider>
  )
}
