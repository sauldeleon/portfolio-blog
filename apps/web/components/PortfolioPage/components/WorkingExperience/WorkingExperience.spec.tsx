import { screen } from '@testing-library/react'

import { renderApp } from '@sdlgr/test-utils'

import { WorkingExperience } from './WorkingExperience'

describe('WorkingExperience', () => {
  it('should render successfully', async () => {
    const { baseElement } = renderApp(<WorkingExperience />)
    expect(
      await screen.findByText(/Part of the team responsible for migrating/),
    ).toBeInTheDocument()

    expect(baseElement).toMatchSnapshot()
  })
})
