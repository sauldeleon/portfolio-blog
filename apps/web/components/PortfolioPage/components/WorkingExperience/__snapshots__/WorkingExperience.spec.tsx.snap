// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkingExperience should render successfully 1`] = `
.c0 {
  font-family: fontFamily;
  font-weight: 400;
  color: #FBFBFB;
  font-size: 1.5rem;
  line-height: normal;
  margin: 0;
}

.c2 {
  font-family: fontFamily;
  font-weight: 500;
  color: #FBFBFB;
  font-size: 1rem;
  line-height: normal;
  text-transform: uppercase;
}

.c1 {
  font-family: fontFamily;
  font-weight: 500;
  margin-bottom: 30px;
}

.c3 {
  font-size: 20px;
  margin-bottom: 20px;
  font-family: fontFamily;
  font-weight: 400;
}

.c4 {
  margin-bottom: 50px;
}

.c5 {
  font-size: 16px;
  text-indent: -20px;
  padding-left: 20px;
}

.c5:before {
  content: 'Â·';
  font-size: 30px;
  vertical-align: middle;
  line-height: 20px;
}

.c6 {
  font-family: fontFamily;
  font-weight: 400;
  color: #FBFBFB;
  font-size: 0.9rem;
  line-height: normal;
  font-size: 16px;
  font-weight: bold;
}

<body>
  <div>
    <div>
      <h1
        class="c0 c1"
      >
        Bonhams
         - 
        Front-end React Developer.
      </h1>
      <div
        class="c2 c3"
      >
        February 2022
      </div>
      <ul
        class="c4"
      >
        <li
          class="c5"
        >
          Part of the team responsible for migrating the legacy website to a cutting-edge 
          <b
            class="c6"
          >
            Next.JS Typescript
          </b>
           application within an 
          <b
            class="c6"
          >
            NX
          </b>
           monorepo. Instrumental in developing a comprehensive Design System from the ground up using 
          <b
            class="c6"
          >
            Styled Components
          </b>
          . Additionally, involved in creating new applications for clients and internal administration, including an 
          <b
            class="c6"
          >
            EXPO
          </b>
           mobile app for iOS and Android.
        </li>
        <li
          class="c5"
        >
          Proficiently employed 
          <b
            class="c6"
          >
            Jest
          </b>
           and 
          <b
            class="c6"
          >
            MSW
          </b>
           for thorough unit and integration testing, while utilizing 
          <b
            class="c6"
          >
            Cypress
          </b>
           for end-to-end testing.
        </li>
        <li
          class="c5"
        >
          Helped at the implementation of a robust 
          <b
            class="c6"
          >
            CI/CD
          </b>
           pipeline using GitLab for efficient and seamless deployment.
        </li>
      </ul>
    </div>
    <div>
      <h1
        class="c0 c1"
      >
        Smart Protection
         - 
        Front-end React Developer.
      </h1>
      <div
        class="c2 c3"
      >
        February 2021 - February 2022
      </div>
      <ul
        class="c4"
      >
        <li
          class="c5"
        >
          Responsible for spearheading the development of back-office applications, catering to the internal administrative requirements of the company's business data.
        </li>
        <li
          class="c5"
        >
          Initiated the creation of a comprehensive monorepo from the ground up using 
          <b
            class="c6"
          >
            Yarn Workspaces
          </b>
          , ensuring the development of full 
          <b
            class="c6"
          >
            TypeScript
          </b>
           applications.
        </li>
        <li
          class="c5"
        >
          Implemented a robust Test-Driven Development (TDD) approach using 
          <b
            class="c6"
          >
            Jest
          </b>
           and 
          <b
            class="c6"
          >
            React Testing Library
          </b>
          .
        </li>
        <li
          class="c5"
        >
          Implemented a sophisticated design system, utilizing a customized library of components using 
          <b
            class="c6"
          >
            Emotion.JS
          </b>
           alongside 
          <b
            class="c6"
          >
            Storybook
          </b>
           for comprehensive documentation.
        </li>
        <li
          class="c5"
        >
          Established and optimized the frontend pipeline in 
          <b
            class="c6"
          >
            Jenkins
          </b>
          , utilizing parallel optimization for multiple stages.
        </li>
        <li
          class="c5"
        >
          Basic knowledge in utilizing Terraform for storing environment variables in AWS.
        </li>
      </ul>
    </div>
    <div>
      <h1
        class="c0 c1"
      >
        ING
         - 
        Front-end Developer.
      </h1>
      <div
        class="c2 c3"
      >
        August 2020 - February 2021
      </div>
      <ul
        class="c4"
      >
        <li
          class="c5"
        >
          Contributed to an international project, focusing on web development based on WebComponents, specifically 
          <b
            class="c6"
          >
            LitElement
          </b>
          , and 
          <b
            class="c6"
          >
            Redux
          </b>
          , with seamless communication with REST APIs.
        </li>
        <li
          class="c5"
        >
          Adopted a Test-Driven Development (TDD) approach using 
          <b
            class="c6"
          >
            Karma
          </b>
           with 
          <b
            class="c6"
          >
            Mocha
          </b>
           for comprehensive unit testing.
        </li>
        <li
          class="c5"
        >
          Conducted end-to-end testing using 
          <b
            class="c6"
          >
            Cucumber
          </b>
           and 
          <b
            class="c6"
          >
            WebdriverIO
          </b>
           for thorough functionality testing.
        </li>
        <li
          class="c5"
        >
          Possesses a strong background in pairing code review experience, serving as an approver of the main repository to ensure the validation of merges into the master branch.
        </li>
      </ul>
    </div>
    <div>
      <h1
        class="c0 c1"
      >
        Keytree
         - 
        Front-end Developer.
      </h1>
      <div
        class="c2 c3"
      >
        October 2017 - August 2020
      </div>
      <ul
        class="c4"
      >
        <li
          class="c5"
        >
          Involved in the development of multiple 
          <b
            class="c6"
          >
            React.JS
          </b>
           web applications, utilizing REST services provided by the Gateway service, primarily based on SAP, Java, or Node.js backend services.
        </li>
        <li
          class="c5"
        >
          Took on the role of a front-end developer lead for an 
          <b
            class="c6"
          >
            Angular JS
          </b>
           application, responsible for onboarding newcomers and overseeing new developments or maintenance. Additionally, proficient in deploying applications across various environments for multiple clients.
        </li>
      </ul>
    </div>
    <div>
      <h1
        class="c0 c1"
      >
        Babel S.I
         - 
        Full Stack Developer.
      </h1>
      <div
        class="c2 c3"
      >
        April 2017 - September 2017
      </div>
      <ul
        class="c4"
      >
        <li
          class="c5"
        >
          Contributed to the Mobile Development Center, focusing on the main project, an 
          <b
            class="c6"
          >
            IONIC
          </b>
           app with 
          <b
            class="c6"
          >
            Angular JS
          </b>
          . Involved in the development of an instant messaging system using 
          <b
            class="c6"
          >
            Socket.io
          </b>
           and 
          <b
            class="c6"
          >
            Node.js
          </b>
          . Integrated 
          <b
            class="c6"
          >
            Spring Boot
          </b>
           and multiple REST services.
        </li>
      </ul>
    </div>
    <div>
      <h1
        class="c0 c1"
      >
        Ioon Technologies
         - 
        Scrum Master and Full Stack Developer.
      </h1>
      <div
        class="c2 c3"
      >
        January 2016 - April 2017
      </div>
      <ul
        class="c4"
      >
        <li
          class="c5"
        >
          Served as the CTO from November 2016 to April 2017. Acted as the Scrum Master of the main I+D+i project, utilizing 
          <b
            class="c6"
          >
            Java 1.7
          </b>
          , 
          <b
            class="c6"
          >
            Spring 4
          </b>
          , 
          <b
            class="c6"
          >
            GWT 1.7
          </b>
          , 
          <b
            class="c6"
          >
            Hibernate 5
          </b>
          , and 
          <b
            class="c6"
          >
            MySQL 5.6
          </b>
          . Implemented 
          <b
            class="c6"
          >
            IONIC 2
          </b>
           and 
          <b
            class="c6"
          >
            Angular
          </b>
           for mobile development.
        </li>
        <li
          class="c5"
        >
          Implemented an 
          <b
            class="c6"
          >
            Arduino
          </b>
           monitoring system for real-time information within the application.
        </li>
        <li
          class="c5"
        >
          Demonstrated proficiency in utilizing 
          <b
            class="c6"
          >
            Maven
          </b>
          , 
          <b
            class="c6"
          >
            Sonarqube
          </b>
          , and 
          <b
            class="c6"
          >
            JUnit
          </b>
          , along with 
          <b
            class="c6"
          >
            Mockito
          </b>
           for testing and validating clean code.
        </li>
        <li
          class="c5"
        >
          Involved in front-end development for various customers, employing 
          <b
            class="c6"
          >
            CSS3
          </b>
          , 
          <b
            class="c6"
          >
            jQuery
          </b>
          , and 
          <b
            class="c6"
          >
            HTML5
          </b>
          .
        </li>
        <li
          class="c5"
        >
          Worked as a 
          <b
            class="c6"
          >
            Java
          </b>
           consultant, focusing on back-end developments in PSA Peugeot-CitroÃ«n. Involved in incident resolution and new functionalities development, utilizing 
          <b
            class="c6"
          >
            Spring 3 MVC
          </b>
          , 
          <b
            class="c6"
          >
            Hibernate
          </b>
          , 
          <b
            class="c6"
          >
            Struts 2
          </b>
           and DB 
          <b
            class="c6"
          >
            Oracle
          </b>
          .
        </li>
      </ul>
    </div>
    <div>
      <h1
        class="c0 c1"
      >
        Clarive
         - 
        Full Stack Developer.
      </h1>
      <div
        class="c2 c3"
      >
        January 2014 - January 2016
      </div>
      <ul
        class="c4"
      >
        <li
          class="c5"
        >
          Involved in back-end development using 
          <b
            class="c6"
          >
            Perl
          </b>
           and the Catalyst Framework, alongside various CPAN modules for seamless database integration, including 
          <b
            class="c6"
          >
            Oracle
          </b>
           and 
          <b
            class="c6"
          >
            MongoDB
          </b>
          .
        </li>
        <li
          class="c5"
        >
          Utilized the 
          <b
            class="c6"
          >
            ExtJS
          </b>
           Framework for front-end development.
        </li>
        <li
          class="c5"
        >
          Integrated SOAP webservices for streamlined communication.
        </li>
      </ul>
    </div>
    <div>
      <h1
        class="c0 c1"
      >
        idealista.com
         - 
        Back-end Developer.
      </h1>
      <div
        class="c2 c3"
      >
        September 2013 - December 2013
      </div>
      <ul
        class="c4"
      >
        <li
          class="c5"
        >
          Involved in back-end development following an Agile (Scrum) methodology.
        </li>
        <li
          class="c5"
        >
          Utilized 
          <b
            class="c6"
          >
            Spring 3
          </b>
           and 
          <b
            class="c6"
          >
            JPA
          </b>
          , incorporating 
          <b
            class="c6"
          >
            Hibernate
          </b>
           with both 
          <b
            class="c6"
          >
            MongoDB
          </b>
           and 
          <b
            class="c6"
          >
            Oracle
          </b>
           databases.
        </li>
      </ul>
    </div>
    <div>
      <h1
        class="c0 c1"
      >
        Incita Security
         - 
        Security Consultant and Web Developer.
      </h1>
      <div
        class="c2 c3"
      >
        November 2012 - September 2013
      </div>
      <ul
        class="c4"
      >
        <li
          class="c5"
        >
          Actively involved in SIEM management, utilizing tools such as ArcSight and OSSIM, and proficient in Shell script development.
        </li>
        <li
          class="c5"
        >
          Demonstrated proficiency in using 
          <b
            class="c6"
          >
            Python
          </b>
           and PHP 5.3, with 
          <b
            class="c6"
          >
            MySQL 5.1
          </b>
           for robust database management.
        </li>
        <li
          class="c5"
        >
          Employed 
          <b
            class="c6"
          >
            CSS3
          </b>
           and 
          <b
            class="c6"
          >
            JavaScript
          </b>
           for efficient and dynamic web development.
        </li>
      </ul>
    </div>
  </div>
</body>
`;
